### PR TITLE
some fixes

### PR DESCRIPTION
**Pull request by joonhocho**
removed timestamp -> property on event object
when user presses enter without selecting suggestion, it will add a tag with the current input value.

@2is10

### CODE REVIEW

File: packrat/scripts/tagbox.js [hunk 1]
```diff
@@ -211,6 +201,11 @@ this.tagbox = (function ($, win) {
 
 			function onKeydown(e) {
 				var preventDefault = false;
+				if (e.metaKey || e.ctrlKey || e.shiftKey) {
```
Comment (2is10) [inline packrat/scripts/tagbox.js:204]: want to include `e.altKey` too?

**Comment by 2is10** [comment]
Nice! I think the event property makes the code clearer. Thanks for doing it.

File: packrat/scripts/tagbox.js (hunk 1 above)
Comment (joonhocho) [inline packrat/scripts/tagbox.js:204]: I copied it from slider2.js. Should I add altKey to both?

**Comment by joonhocho** [comment]
I agree that using event property is a better solution. I couldn't think of that at the time. It is a good way to communicate between event handlers.

File: packrat/scripts/tagbox.js (hunk 1 above)
Comment (2is10) [inline packrat/scripts/tagbox.js:204]: Sure, thanks.

I know several OSes have [default key bindings for Esc](http://en.wikipedia.org/wiki/Table_of_keyboard_shortcuts#Navigation) with various modifiers, but it intercepts them before sending them to applications, so I think we only need to concern ourselves with possible conflicts in web pages/apps and perhaps mis-typing of OS shortcuts. Best to ignore all modifier keys, I think.

File: packrat/scripts/tagbox.js [hunk 2]
```diff
@@ -259,6 +240,19 @@ this.tagbox = (function ($, win) {
...
 
+		handleEsc: function (e) {
+			log('handle esc');
+			e.preventDefault();
+			e.stopPropagation();
+			e.stopImmediatePropagation();
+			if (this.currentSuggestion) {
+				this.navigateTo(null, 'esc');
+			}
+			else {
```
Comment (2is10) [inline packrat/scripts/tagbox.js:251]: We’ve been using the `} else {` style to conserve vertical space.

**Pull request merged**